test/e2e/jobcleanup_test.go

```diff
@@ -0,0 +1,127 @@
+package e2e
+
+import (
+	"github.com/keptn/go-utils/pkg/api/models"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+func TestJobCleanupWithSmallTTL(t *testing.T) {
+	if !isE2ETestingAllowed() {
+		t.Skip("Skipping TestJobCleanupWith0TTLMultipleJobs, not allowed by environment")
+	}
+
+	testEnv := setupE2ETTestEnvironment(t,
+		"../events/e2e/jobcleanup.triggered.json",
+		"../shipyard/e2e/jobcleanup.deployment.yaml",
+		"../data/e2e/jobcleanup.config.yaml",
+	)
+
+	defer testEnv.CleanupFunc()
+
+	// Send the event to keptn
+	keptnContext, err := testEnv.API.SendEvent(testEnv.Event)
+	require.NoError(t, err)
+
+	// Checking if the job executor service responded with a .started event
+	requireWaitForEvent(t,
+		testEnv.API,
+		2*time.Minute,
+		1*time.Second,
+		keptnContext,
+		"sh.keptn.event.deployment.started",
+		func(_ *models.KeptnContextExtendedCE) bool {
+			return true
+		},
+	)
+
+	// If the started event was sent by the job executor we wait for a .finished with the following data:
+	expectedEventData := eventData{
+		Project: testEnv.EventData.Project,
+		Result:  "pass",
+		Service: testEnv.EventData.Service,
+		Stage:   testEnv.EventData.Stage,
+		Status:  "succeeded",
+	}
+
+	requireWaitForEvent(t,
+		testEnv.API,
+		2*time.Minute,
+		1*time.Second,
+		keptnContext,
+		"sh.keptn.event.deployment.finished",
+		func(event *models.KeptnContextExtendedCE) bool {
+			responseEventData, err := parseKeptnEventData(event)
+			require.NoError(t, err)
+
+			t.Log(responseEventData.Message)
+
+			responseEventData.Message = ""
+
+			assert.Equal(t, expectedEventData, *responseEventData)
+			return true
+		},
+	)
+}
+
+func TestJobCleanupWith0TTLMultipleJobs(t *testing.T) {
+	if !isE2ETestingAllowed() {
+		t.Skip("Skipping TestJobCleanupWith0TTLMultipleJobs, not allowed by environment")
+	}
```

Comment:
Since this is currently not working (untile the fix for TTL 0 is added, I would modify it as follows

```suggestion
        t.Skip("Skipping TestJobCleanupWith0TTLMultipleJobs since TTL=0 is not currently supported/working")
	if !isE2ETestingAllowed() {
		t.Skip("Skipping TestJobCleanupWith0TTLMultipleJobs, not allowed by environment")
	}
```